frontend/packages/console-plugin-sdk/src/store.ts

```diff
@@ -0,0 +1,69 @@
+import * as _ from 'lodash';
+import {
+  Extension,
+  AlwaysOnExtension,
+  ExtensionWithMetadata,
+  ActivePlugin,
+  isFeatureFlag,
+  isModelDefinition,
+} from './typings';
+import { ExtensionRegistry } from './registry';
+
+/**
+ * Always-on extensions are ones whose declared type extends `AlwaysOnExtension`.
+ */
+export const isAlwaysOnExtension = (e: Extension): e is AlwaysOnExtension => {
+  return isFeatureFlag(e) || isModelDefinition(e);
+};
+
+export const isGatedExtension = (e: Extension): boolean => !isAlwaysOnExtension(e);
+
+export const sanitizeExtension = (e: Extension): Extension => {
+  if (isGatedExtension(e)) {
+    e.flags = e.flags || {};
+    e.flags.required = _.uniq(e.flags.required || []);
+    e.flags.disallowed = _.uniq(e.flags.disallowed || []);
+  } else if (e.flags !== undefined) {
+    // eslint-disable-next-line no-console
+    console.warn('Discarding flags for always-on extension:', e);
+    delete e.flags;
+  }
+  return e;
+};
+
+export const augmentExtension = (e: Extension, p: ActivePlugin): ExtensionWithMetadata => {
+  return Object.assign(e, { plugin: p.name });
+};
+
+export const isExtensionInUse = (e: Extension, flags: { [key: string]: boolean }): boolean => {
+  const isPending = (f: string) => flags[f] === undefined;
```

Comment:
FYI, can't use `flagPending()` or `FlagsObject` from `public/reducers/features.ts` as it causes Jest to fail on cyclic dependency.